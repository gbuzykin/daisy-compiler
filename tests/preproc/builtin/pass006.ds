#define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)

const a = SUM  (1, 2);
const b = SUM  (1, 2, 3 + 4);
