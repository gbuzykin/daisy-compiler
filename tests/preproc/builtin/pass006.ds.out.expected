./preproc/builtin/pass006.ds:3:1: debug: token
 3 | const a = SUM  (1, 2);
   | ^~~~~
./preproc/builtin/pass006.ds:3:7: debug: id: a
 3 | const a = SUM  (1, 2);
   |       ^
./preproc/builtin/pass006.ds:3:9: debug: token
 3 | const a = SUM  (1, 2);
   |         ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:24: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                        ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:25: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                         ^
./preproc/builtin/pass006.ds:3:17: debug: integer number: 1
 3 | const a = SUM  (1, 2);
   |                 ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:27: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                           ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:29: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                             ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:31: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                               ^
./preproc/builtin/pass006.ds:3:20: debug: integer number: 2
 3 | const a = SUM  (1, 2);
   |                    ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:33: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                                 ^
./preproc/builtin/pass006.ds:3:11: debug: token
 3 | const a = SUM  (1, 2);
   |           ^~~~~~~~~~~
./preproc/builtin/pass006.ds:1:60: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                                                            ^
./preproc/builtin/pass006.ds:3:22: debug: token
 3 | const a = SUM  (1, 2);
   |                      ^
./preproc/builtin/pass006.ds:4:1: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   | ^~~~~
./preproc/builtin/pass006.ds:4:7: debug: id: b
 4 | const b = SUM  (1, 2, 3 + 4);
   |       ^
./preproc/builtin/pass006.ds:4:9: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |         ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:24: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                        ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:25: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                         ^
./preproc/builtin/pass006.ds:4:17: debug: integer number: 1
 4 | const b = SUM  (1, 2, 3 + 4);
   |                 ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:27: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                           ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:29: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                             ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:31: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                               ^
./preproc/builtin/pass006.ds:4:20: debug: integer number: 2
 4 | const b = SUM  (1, 2, 3 + 4);
   |                    ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:33: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                                 ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:46: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                                              ^
./preproc/builtin/pass006.ds:4:23: debug: integer number: 3
 4 | const b = SUM  (1, 2, 3 + 4);
   |                       ^
./preproc/builtin/pass006.ds:4:25: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |                         ^
./preproc/builtin/pass006.ds:4:27: debug: integer number: 4
 4 | const b = SUM  (1, 2, 3 + 4);
   |                           ^
./preproc/builtin/pass006.ds:4:11: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |           ^~~~~~~~~~~~~~~~~~
./preproc/builtin/pass006.ds:1:60: note: expanded from macro `SUM`
 1 | #define SUM(a, b, ...) ((a) + (b) __va_opt__(+) __va_args__)
   |                                                            ^
./preproc/builtin/pass006.ds:4:29: debug: token
 4 | const b = SUM  (1, 2, 3 + 4);
   |                             ^
./preproc/builtin/pass006.ds: info: warnings 0, errors 0
